core/Tracker/LogTable.php

```diff
@@ -120,4 +120,15 @@ public function getPrimaryKey()
     {
         return array();
     }
+
+    /**
+     * Defines if the log table holds an `idvisitor` column.
+     * If set to yes, existing records are automatically updated when the visitor id changes during a visit
+     *
+     * @return bool
+     */
+    public function hasIdVisitorColumn(): bool
```

Comment:
If we named this e.g. `hasDependentIdVisitorColumn` or `hasSecondaryIdVisitorColumn` or similar to indicate it either doesn't have the column or it does and it shouldn't be updated, we could then skip the check for the instance of the Visit table `|| $logTable instanceof \Piwik\Plugins\CoreHome\Tracker\LogTable\Visit` below and rely only on this method's return value. Just a thought for consideration, the exception for a single table in the model class is acceptable.